helix-term/src/keymap/default.rs

```diff
@@ -163,7 +193,6 @@ pub fn default() -> HashMap<Mode, Keymap> {
 
         "A-:" => ensure_selections_forward,
 
-        "esc" => normal_mode,
```

Comment:
When we get plugins, it may be good to still trigger the `normal_mode`, so plugins can react to the fact that the user wants to opt out (close modals or whatever). Unless the plan is for the modals/plugins to all specify their escape much like the current ones do such as space k.
Just my immediate thoughts.